lambdas/questions/src/main/java/uk/gov/di/ipv/cri/experian/kbv/api/QuestionHandler.java

```diff
@@ -1,139 +1,92 @@
-            InputValidationExecutor inputValidationExecutor) {
+            KeyStoreService keyStore,
+            KBVServiceFactory kbvServiceFactory,
+            InputValidationExecutor inputValidationExecutor
+    )  {
         this.inputValidationExecutor = inputValidationExecutor;
-        this.kbvService = kbvService;
         this.objectMapper = objectMapper;
+        this.kbvService = kbvServiceFactory.create();
+        this.objectMapper.registerModule(new JavaTimeModule());
+        this.response = new APIGatewayProxyResponseEvent();
+
+        persistKeyStoreToSystemProperty(keyStore.getValue(), keyStore.getPassword());
     }
 
     @Override
     @Tracing(captureMode = CaptureMode.DISABLED)
     public APIGatewayProxyResponseEvent handleRequest(
             APIGatewayProxyRequestEvent input, Context context) {
-        Map<String, String> responseHeaders = Map.of("Content-Type", "application/json");
-        int responseStatusCode;
-        String responseBody;
 
+        response.withHeaders(Map.of("Content-Type", "application/json"));
         try {
-            PersonIdentity personIdentity =
-                    this.objectMapper.readValue(input.getBody(), PersonIdentity.class);
-
+            QuestionRequest questionRequest =
+                    this.objectMapper.readValue(input.getBody(), QuestionRequest.class);
             ValidationResult validationResult =
-                    this.inputValidationExecutor.performInputValidation(personIdentity);
+                    this.inputValidationExecutor.performInputValidation(questionRequest.getPersonIdentity());
 
             if (validationResult.isValid()) {
-                QuestionsResponse questionsResponse = this.kbvService.getQuestions(personIdentity);
-                responseStatusCode = 200;
-                responseBody = objectMapper.writeValueAsString(questionsResponse);
+                QuestionsResponse questionsResponse = this.kbvService.getQuestions(questionRequest);
+                response.withStatusCode(HttpStatus.SC_OK);
+                response.withBody(objectMapper.writeValueAsString(questionsResponse));
             } else {
-                responseStatusCode = 400;
-                responseBody = objectMapper.writeValueAsString(validationResult);
+                response.withStatusCode(HttpStatus.SC_BAD_REQUEST);
+                response.withBody(objectMapper.writeValueAsString(validationResult));
             }
         } catch (Exception e) {
-            LOGGER.error("Error occurred when attempting to retrieve questions", e);
-            responseStatusCode = 500;
-            responseBody = "{\"error\": \"" + e + "\"}";
+            LambdaLogger logger = context.getLogger();
+            logger.log(String.format("Error occurred when attempting to retrieve questions %s", e));
+            response.withStatusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+            response.withBody("{\"error\": \"" + e + "\"}");
```

Comment:
it would be worthwhile exposing `e.getMessage()` or something a bit more succinct as opposed to the entire exception here